src/Caching/SqlServer/src/DatabaseOperations.cs

```diff
@@ -238,8 +239,10 @@ protected virtual byte[] GetCacheItem(string key, bool includeValue)
 
                         if (!reader.IsDBNull(Columns.Indexes.AbsoluteExpirationIndex))
                         {
-                            absoluteExpiration = reader.GetFieldValue<DateTimeOffset>(
+                            var absoluteExpirationStr = reader.GetFieldValue<object>(
                                 Columns.Indexes.AbsoluteExpirationIndex);
+                            DateTimeOffset.TryParse(absoluteExpirationStr?.ToString(), out var currentAbsoluteExpiration);
```

Comment:
Is it save to convert to string representation?  Won't this result in loss of information like time-zone?

Reply:
It is save and never loss relate  information of datetime.

Reply:
@tarekgh captured my concern here: https://github.com/dotnet/extensions/pull/2791#discussion_r504328413